### ISSUE TITLE
second error: SBCL disconnected

### ISSUE THREAD

**Issue by vindarel** [Ragnaroek/dandelion#1]
Hello,

First of all… did you really solve the number 1 complain about Common Lisp, "there is no decent editor besides Emacs" ? O_o

I couldn't test Dandelion yet. When evaluating the file I got 3 errors, I got passed one by issuing a `chmod +x` as you explained:

    chmod +x /home/vince/.p2/pool/plugins/de.defmacro.dandelion.env.sbcl.mac.intel_1.1.10/binary/environment_sbcl_1.1.10

(along with the other binaries I could find, including the linux one)

but I still get 2, and SBCL is disconnected:

![selection_265](https://user-images.githubusercontent.com/3721004/50060063-1f625580-018f-11e9-80cf-8aa22039d8e6.png)

> Error initialising connection
java.net.ConnectException: Connexion refusée (Connection refused)

Is there something else I can do ?

Thanks

ps: as a side request, would you furnish the readme with a tad more indications of what this plugin do or doesn't ? (what about an interactive debugger ?) This is worth *two* beers :p 

pps: your are now live on [reddit](https://www.reddit.com/r/Common_Lisp/comments/a5t3xx/dandelion_eclipse_common_lisp_editor_last_commit/) and will be soon in the [cookbook](https://github.com/LispCookbook/cl-cookbook/pull/194) and the [awesome list](https://github.com/CodyReichert/awesome-cl/pull/221). And I humbly invite you to give some news when you can on reddit. This plugin went unnoticed for too long (even though it has decent stats on the market place, 1.5k downloads, 86 last month).

and while I'm at it… did you use some CUSP code or not at all ? (reddit question)

**Comment by Ragnaroek** [comment]
Probably the binary part of the editor is not coming up properly. Can you try to start the binary at
/home/vince/.p2/pool/plugins/de.defmacro.dandelion.env.sbcl.mac.intel_1.1.10/binary/environment_sbcl_1.1.10 directly and post the error message?

You are on MacOS? If you install all binaries, you have to be careful selecting the right one for your platform.

**Comment by vindarel** [comment]
I am on Linux (LMDE Debian 8 x64), but the Mac one is marked as required.

![selection_263](https://user-images.githubusercontent.com/3721004/50118160-26e93380-024f-11e9-8074-3f1de29ea347.png)

I ran Eclipse in the terminal and saw a little stacktrace when trying to evaluate a file: 

> DEBUG: Process died
DEBUG err=/home/vince/.eclipse/rust-2018-09/eclipse/../../../.p2/pool/plugins/de.defmacro.dandelion.env.sbcl.mac.intel_1.1.10/binary/environment_sbcl_1.1.10: 1: /home/vince/.eclipse/rust-2018-09/eclipse/../../../.p2/pool/plugins/de.defmacro.dandelion.env.sbcl.mac.intel_1.1.10/binary/environment_sbcl_1.1.10: /home/vince/.eclipse/rust-2018-09/eclipse/../../../.p2/pool/plugins/de.defmacro.dandelion.env.sbcl.mac.intel_1.1.10/binary/environment_sbcl_1.1.10: 1: /home/vince/.eclipse/rust-2018-09/eclipse/../../../.p2/pool/plugins/de.defmacro.dandelion.env.sbcl.mac.intel_1.1.10/binary/environment_sbcl_1.1.10: $: not found��__unwind_info__TEXT���__eh_frame__TEXT���G��x__DATA�: not found

it runs the Mac one.

and running the linux binary (with a port) seems fine:

> $ /home/vince/.p2/pool/plugins/de.defmacro.dandelion.env.sbcl.linux.x86_1.1.10/binary/environment_sbcl_1.1.10 44000
> SERVER_UP

I'd like to connect to it but I don't know how.

**Comment by Ragnaroek** [comment]
Can you select the linux environment for your project. It should be configurable in the project settings and the listener view.

Having the Mac-Version required on Linux is a bug.

**Comment by vindarel** [comment]
with

    $ /home/vince/.p2/pool/plugins/de.defmacro.dandelion.env.sbcl.linux.x86_1.1.10/binary/environment_sbcl_1.1.10 30240

I can evaluate a file and see results in the REPL a couple of times, but it errors out after a short while:

![selection_263](https://user-images.githubusercontent.com/3721004/50233094-f3271e80-03b2-11e9-80c6-6b27d3a45f2d.png)

I see this on the shell. It appears a few seconds after the time out (of ±20s), even when the file is correct:

```
<ERROR> [17:25:26] dandelion-server server.lisp (start-handle-loop) -
  handle loop unexpected exited: end of file on #<SB-SYS:FD-STREAM
                                                  for "socket 127.0.0.1:30240, peer: 127.0.0.1:59314"
                                                  {B6E97F9}>
CORRUPTION WARNING in SBCL pid 31405(tid 4158338816):
Memory fault at b7fd3000 (pc=0x952d8c9, sp=0xf78e1ec0)
The integrity of this image is possibly compromised.
Continuing with fingers crossed.

debugger invoked on a SB-SYS:MEMORY-FAULT-ERROR in thread
#<THREAD "main thread" RUNNING {B322371}>:
  Unhandled memory fault at #xB7FD3000.

Type HELP for debugger help, or (SB-EXT:EXIT) to exit from SBCL.

(no restarts: If you didn't do this on purpose, please report it as a bug.)

(SB-SYS:MEMORY-FAULT-ERROR)
0] 
```

> Can you select the linux environment for your project.

ok I found the Preferences, there I only see one SBCL, don't know which one it is and I suppose it isn't the Linux one:

![selection_264](https://user-images.githubusercontent.com/3721004/50233194-2ec1e880-03b3-11e9-8d25-c133672d1629.png)

**Comment by Ragnaroek** [comment]
Ok, it is not fully broken.

The "Continuing with fingers crossed." and "Unhandled memory fault at #xB7FD3000." doesn't look very promising, though. I think the correct binary is selected when the listener is started (otherwise the binary wouldn't start because of wrong architecture).

Anyhow, I have to debug this (install Eclipse, try it myself on my machine). I will post results here.

**Comment by Ragnaroek** [comment]
Could you post the code you are trying to evaluate?

On a first (shallow) try-out it works on my Mac.

**Comment by vindarel** [comment]
sure, this simple code:

```
(format t "hello from Eclipse !~&")

(defclass person ()
   ((name :initarg :name :accessor name)))

(defvar *p* (make-instance 'person :name "me"))

(print "person:")	
(print *p*)
```


I installed Eclips 2018-09 (4.9.0).

**Comment by Ragnaroek** [comment]
Works on MacOS. I will retry this on Linux (but have to install it first :)

**Comment by Ragnaroek** [comment]
On Linux it tries to run the MacOS binary, can confirm that on my installation.

Can you try to delete the MacoOS plugins (Help > About Eclipse IDE > Installation Details button.
'Uninstall' all 'Dandelion MacOS' plugins in the list.
Restart Eclipse.

Can you send me what happens if you do that?

**Comment by vindarel** [comment]
I can evaluate code :) 

At startup the terminal says "SBCL disconnected", and after a first evaluation it shows "connected".

However, I evaluated some code with an error, after shutting down Eclipse I still have a `environment_sbcl` running and showing a debugger. It seems to be running in the background (I do have a shell prompt, I see the process with ps), but typing in the shell fires the debugger. So I can't type `killall envir…`: 

```
$ kil                                                                                        *[master] 

debugger invoked on a SB-INT:SIMPLE-STREAM-ERROR in thread
#<THREAD "main thread" RUNNING {B322399}>:
  couldn't read from #<SB-SYS:FD-STREAM for "the terminal" {B3240E1}>:
    Erreur d'entrée/sortie

Type HELP for debugger help, or (SB-EXT:EXIT) to exit from SBCL.

(no restarts: If you didn't do this on purpose, please report it as a bug.)

(SB-IMPL::SIMPLE-STREAM-PERROR
 "couldn't read from ~S"
 #<SB-SYS:FD-STREAM for "the terminal" {B3240E1}>
 5)
0[10] 
```

when I press enter I get the shell prompt again, and same story.

I could reproduce. It seems the shell doesn't accept more than 2 characters.

**Comment by Ragnaroek** [comment]
Can you try to update. I update the SBCL Linux environment to a more recent 1.4.10 version. Maybe that fixes something. If not, we know that it is not some outdated environment problem.

**Comment by vindarel** [comment]
Finally found how to update with Help > check for updates.

It works better, because the form that threw an error (but was correct in Slime) doesn't throw one anymore. I still have the weird behavior after shutting down Eclipse. 

Thanks already for the update (and all).

**Comment by vindarel** [comment]
ps: a feedback on the readme: you could add that Dandelion has an interactive debugger, list other features, have more high-res srceenshots and give more clues for the ones who don't know Eclipse. (I might make a little PR about next week but you know better about the features)

**Comment by Ragnaroek** [comment]
Good that it is working better for you now.

I digged a little through the code, there doesn't seem to be something for a orderly shutdown, when Eclipse exits. This is definitely something that should be there, but probably never got implemented.

Yes, the Readme needs some revamping. I will work on that.
But I have to disappoint you, Dandelion does not have an interactive debugger (it only has a dedicated apropos view).
